pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -663,6 +667,35 @@ func GetShortNameMode(ctx *types.SystemContext) (types.ShortNameMode, error) {
 	return config.shortNameMode, err
 }
 
+func subdomainMatchPrefix(ref, prefix string) bool {
```

Comment:
My Friday-afternoon brain has trouble reading the code.  Couldn't we use `strings.HasSuffix(ref, prefix[2:])`?

Reply:
Ah no ... that obviously won't work.

Reply:
`strings.Index(...)` maybe?